File: backend/src/main/java/magazyn/dto/RaportFinansowyDTO.java
package magazyn.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class RaportFinansowyDTO {

    private LocalDateTime dataPoczatek;
    private LocalDateTime dataKoniec;
    private BigDecimal sumaPrzychodow;
    private BigDecimal sumaWydatkow;
    private BigDecimal sumaZysku;
    private Integer liczbaTransakcji;
    private String typ;

    public RaportFinansowyDTO(LocalDateTime dataPoczatek, LocalDateTime dataKoniec, 
                              BigDecimal sumaPrzychodow, BigDecimal sumaWydatkow, BigDecimal sumaZysku) {
        this.dataPoczatek = dataPoczatek;
        this.dataKoniec = dataKoniec;
        this.sumaPrzychodow = sumaPrzychodow != null ? sumaPrzychodow : BigDecimal.ZERO;
        this.sumaWydatkow = sumaWydatkow != null ? sumaWydatkow : BigDecimal.ZERO;
        this.sumaZysku = sumaZysku != null ? sumaZysku : BigDecimal.ZERO;
    }
}

File: frontend/app/src/main/java/com/example/magazyn/api/dtos/RaportFinansowyDTO.kt
package com.example.magazyn.api.dtos

import java.math.BigDecimal

data class RaportFinansowyDTO(
    val dataPoczatek: String? = null,
    val dataKoniec: String? = null,
    val sumaPrzychodow: BigDecimal = BigDecimal.ZERO,
    val sumaWydatkow: BigDecimal = BigDecimal.ZERO,
    val sumaZysku: BigDecimal = BigDecimal.ZERO,
    val liczbaTransakcji: Int = 0,
    val typ: String = "ALL"
)
